src/common.rs

```diff
@@ -85,49 +83,157 @@ pub fn append_output(text: String, file: &str, o: &mut Outputs) {
+        return Some(pat);
+    } else {
+        return None;
+    }
 }
 
 /// Check the function argument is `user_data: *mut c_void`
-pub fn is_user_data_arg(arg: &ast::Arg) -> bool {
-    pprust::pat_to_string(&*arg.pat) == "user_data"
-        && pprust::ty_to_string(&*arg.ty) == "*mut c_void"
+pub fn is_user_data_arg(arg: &syn::ArgCaptured) -> bool {
+    let mut flags = (0, 0);
+    if let syn::Pat::Ident(ref pat) = arg.pat {
+        if pat.to_owned().ident.to_string().as_str() == "user_data" {
+            flags.0 = 1;
+        }
+    }
+    if let syn::Type::Ptr(ref pat) = arg.ty {
+        if pat.to_owned().into_token_stream().to_string().as_str() == "* mut c_void" {
+            flags.1 = 1;
+        }
+    }
+    if flags == (1, 1) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+pub fn is_user_data_arg_barefn(arg: &syn::BareFnArg) -> bool {
+    let mut flags = (0, 0);
+    if unwrap!(arg.to_owned().name)
+        .0
+        .to_owned()
+        .into_token_stream()
+        .to_string()
+        .to_owned()
+        .as_str()
+        == "user_data"
+    {
+        flags.0 = 1;
+    }
+    if let syn::Type::Ptr(ref pat) = arg.ty {
+        if pat.to_owned().into_token_stream().to_string().as_str() == "* mut c_void" {
+            flags.1 = 1;
+        }
+    }
+    if flags == (1, 1) {
+        return true;
+    } else {
+        return false;
+    }
 }
 
 /// Check the function argument is `result: *const FfiResult`
-pub fn is_result_arg(arg: &ast::Arg) -> bool {
-    pprust::pat_to_string(&*arg.pat) == "result"
-        && pprust::ty_to_string(&*arg.ty) == "*const FfiResult"
+pub fn is_result_arg(arg: &syn::ArgCaptured) -> bool {
+    let mut flags = (0, 0);
+    if let syn::Pat::Ident(ref pat) = arg.pat {
+        if pat.ident.to_owned().to_string().as_str() == "result" {
+            flags.0 = 1;
+        } else {
+            flags.0 = 0;
+        }
+    }
+    if let syn::Type::Ptr(ref ptr) = arg.ty {
+        if ptr.to_owned().into_token_stream().to_string().as_str() == "*const FfiResult" {
+            flags.1 = 1;
+        } else {
+            flags.0 = 0;
+        }
+    }
+    if flags == (1, 1) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+pub fn is_result_arg_barefn(arg: &syn::BareFnArg) -> bool {
+    let mut vector = vec![];
+    for vec in arg.to_owned().into_token_stream().to_string().split(":") {
+        vector.push(format!("{}", &vec));
+    }
+    if unwrap!(vector.first()).as_str() == "result "
+        && unwrap!(vector.last()).as_str() == "*const FfiResult"
+    {
+        return true;
+    } else {
+        return false;
+    }
 }
 
 /// Check the function argument is a length argument for a *const u8 pointer
-pub fn is_ptr_len_arg(ty: &ast::Ty, arg_name: &str) -> bool {
-    pprust::ty_to_string(ty) == "usize"
+pub fn is_ptr_len_arg(ty: &syn::Type, arg_name: &str) -> bool {
+    &*ty.to_owned().into_token_stream().to_string().as_str() == "usize"
```

Comment:
I think it should be also possible to simplify this as:
```suggestion
    &*ty.into_token_stream().to_string() == "usize"
```